src/bch/coin.js

```diff
@@ -102,15 +103,17 @@ Coin.TX_INPUT_PUBKEYHASH = 106;
 Coin.TX_OUTPUT_BASE = 8 + 1;
 Coin.TX_OUTPUT_PUBKEYHASH = 25;
 
-Coin.TX_INPUT_SEGWIT_DUST_SERVICE = 130
-Coin.TX_OUTPUT_SEGWIT_DUST_SERVICE = 0
+Coin.TX_INPUT_SEGWIT_DUST_SERVICE = 130;
+Coin.TX_OUTPUT_SEGWIT_DUST_SERVICE = 0;
 
 Coin.empty = Coin.of(0);
 
 Coin.inputBytes = (_input) => _input.dust ? Coin.TX_INPUT_SEGWIT_DUST_SERVICE : Coin.TX_INPUT_BASE + Coin.TX_INPUT_PUBKEYHASH;
 
 Coin.outputBytes = (_output) => _output.dust ? Coin.TX_OUTPUT_SEGWIT_DUST_SERVICE : Coin.TX_OUTPUT_BASE + Coin.TX_OUTPUT_PUBKEYHASH;
 
-Coin.effectiveValue = curry((feePerByte, coin) => coin.value - feePerByte * Coin.inputBytes(coin));
+Coin.effectiveValue = curry((feePerByte, coin) =>
+  clamp(0, Infinity, coin.value - feePerByte * Coin.inputBytes(coin))
```

Comment:
This wasn't clamped before, which allowed for negative values. Was that intentional? (It was causing a test to fail)